src/deluge/io/midi/midi_engine.cpp

```diff
@@ -82,8 +82,24 @@ void usbSendComplete(int ip) {
 	bool inHostMode = (g_usb_usbmode == USB_HOST);
 
 	// If in host mode, see if we want to send to another device
+	// NB: this only gets called in host mode anyway, refactor!
 	if (inHostMode) {
 
+		// check if there was more to send on the same device
+		bool has_more = connectedDevice->consumeBytes();
+		if (has_more) {
+			// TODO: do some cooperative scheduling here. so if there is a flood of data
+			// on connected device 1 and we just want to send a few notes on device 2,
+			// make sure device 2 ges a fair shot now and then
+
+			g_usb_midi_send_utr[USB_CFG_USE_USBIP].tranlen = connectedDevice->numBytesSendingNow;
+			g_usb_midi_send_utr[USB_CFG_USE_USBIP].p_tranadr = connectedDevice->dataSendingNow;
+			int pipeNumber = g_usb_hmidi_tmp_ep_tbl[USB_CFG_USE_USBIP][midiDeviceNum][0];
+			usb_send_start_rohan(&g_usb_midi_send_utr[USB_CFG_USE_USBIP], pipeNumber, connectedDevice->dataSendingNow,
+			                     connectedDevice->numBytesSendingNow);
+			return;
```

Comment:
How does this interact with the previous code to flush all devices? Is that now unreachable?

Reply:
it still works with the caveat mentioned in the comment above. I might restructure this function to make it clearer as it is now explicit that this function is only used for host mode.